### PR TITLE
feat: restperf Timestamp handling per batch

### PR DESCRIPTION
**Pull request by rahulguptajss**

### CODE REVIEW

File: cmd/tools/rest/rest.go [hunk 1]
```diff
@@ -180,12 +185,14 @@ func doData() {
 	}
 
 	// strip leading slash
-	args.API = strings.TrimPrefix(args.API, "/")
+	if strings.HasPrefix(args.API, "/") {
```
Comment (cgrinds) [inline cmd/tools/rest/rest.go:188]: the previous version is clearer `args.API = strings.TrimPrefix(args.API, "/")`

File: cmd/tools/rest/rest.go [hunk 2]
```diff
@@ -270,6 +277,43 @@ func FetchData(client *Client, href string, records *[]any) error {
 	return nil
 }
 
+// FetchRestPerfData THis method is used in PerfRest collector. This method returns timestamp per batch
```
Comment (cgrinds) [inline cmd/tools/rest/rest.go:280]: minor: `This` instead of `THis`

**Review by cgrinds — COMMENTED** [review]
let's discuss tomorrow, I've got a question about the flow

File: cmd/collectors/restperf/restperf.go [hunk 3]
```diff
@@ -511,205 +494,232 @@ func (r *RestPerf) PollData() (map[string]*matrix.Matrix, error) {
...
-		var (
-			instanceKey string
-			instance    *matrix.Instance
-		)
+	numRecords := 0
 
-		if !instanceData.IsObject() {
-			r.Logger.Warn().Str("type", instanceData.Type.String()).Msg("Instance data is not object, skipping")
-			return true
+	instanceKeys = r.Prop.InstanceKeys
```
Comment (Hardikl) [inline cmd/collectors/restperf/restperf.go:499]: this should be before 484 line, else overwritten.

File: cmd/tools/rest/rest.go [hunk 4]
```diff
@@ -270,7 +276,44 @@ func FetchData(client *Client, href string, records *[]any) error {
...
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func stderr(format string, a ...any) {
```
Comment (Hardikl) [inline cmd/tools/rest/rest.go:316]: we need to remove line 332 as we always do downloadAll

**Pull request merged**